src/lib/game-saving/account.ts

```diff
@@ -331,4 +336,27 @@ export const getSnapshotData = async (id: string): Promise<SnapshotData | null>
 		ownerName: user?.username ?? snapshot.ownerName,
 		code: snapshot.code
 	}
-}
\ No newline at end of file
+}
+
+export const updateUserGitHubToken = async (userId: string, githubAccessToken: string, githubId: string, githubUsername: string): Promise<void> => {
+    await updateDocument('users', userId, { githubAccessToken, githubId, githubUsername });
+}
+
+async function hashCodeToBigInt(string : string) : Promise<bigint>{
+	return BigInt(`0x${ await sha256Hash(string)}`);
+}
+
+export async function isAccountWhitelistedToUseCollabAndSavingBetaFeatures(id: string, email: string) : Promise<boolean>{
+	if(import.meta.env.PERCENT_OF_USERS_WHITELISTED_FOR_BETA_FEATURE == 0 || import.meta.env.PERCENT_OF_USERS_WHITELISTED_FOR_BETA_FEATURE == undefined) return false;
+	let hashedId = await hashCodeToBigInt(id);
+	
+	if(hashedId % BigInt(100) < import.meta.env.PERCENT_OF_USERS_WHITELISTED_FOR_BETA_FEATURE || 
```

Comment:
Good that you modified this to be percentage based, so much easier to understand now.